cmd/juju/application/utils/origin.go

```diff
@@ -56,9 +52,11 @@ func DeduceOrigin(url *charm.URL, channel charm.Channel, platform corecharm.Plat
 			Branch:       branch,
 			Architecture: architecture,
 		}
+	default:
+		return commoncharm.Origin{}, errors.NotSupportedf("charm source %q", schema)
 	}
-	if url.Revision != -1 {
-		origin.Revision = &url.Revision
+	if revision != -1 {
```

Comment:
```suggestion
	if revision >= 0 {
```

Reply:
Disagree with this suggestion. We use `revision != -1` **everywhere**.  Even just this PR still uses that elsewhere. Let's stick to accepted convention. Revision `-1` has universally been used to date to mean "not specified". If we wanted to introduce a `const UnspecifiedRevision = -1` at some point and compare to that, and update **all** the call sites, then fine. But I don't like changing just one place in the code and leaving things inconsistent.